src/librustc_metadata/diagnostics.rs

```diff
@@ -26,6 +26,18 @@ name. Example:
 ```
 "##,
 
+E0455: r##"
+A framework link kind was used and the OS utilized to compile isn't the OSX.
+Erroneous code example:
+
+```compile_fail"
+#[link(name = "FooCoreServices",  kind = "framework")] extern {}
+// OS used to compile is Linux for example
+```
+
+When using 'framework' link kind ensure to utilize OSX to compile the code.
```

Comment:
"When using 'framework' link kind, please ensure to use OSX to compile the code."